### PR TITLE
Add ci job for formatting and linting

### PR DESCRIPTION
**Pull request by smlabt**

### CODE REVIEW

File: .github/workflows/pr-validation.yml [hunk 1]
```diff
@@ -34,6 +34,30 @@ jobs:
...
+
+      - name: Format and Lint all files
+        working-directory: ./ui
+        run: |
+          npm run format
+          npm run lint
+
   run-cypress-e2e-tests:
     runs-on: ubuntu-latest
     needs: [python-quality-checks, python-unit-tests]
```
Comment (bossenti) [inline .github/workflows/pr-validation.yml:63]: please add `run-format-and-lint-ui` here

File: ui/.prettierignore [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+dist
+.angular
+
+# Remove these in the future to format additional modules
+# Please also see .eslintignore
+cypress
```
Comment (bossenti) [inline ui/.prettierignore:21]: I have created #877 to track the progress

**Review by bossenti — COMMENTED** [review]
awesome @smlabt, thanks a lot!

File: .github/workflows/pr-validation.yml (hunk 1 above)
Comment (smlabt) [reply .github/workflows/pr-validation.yml]: Done! :)

**Review by bossenti — APPROVED** [review]
From my point of view we are ready to merge :) 
@smlabt anything open from your side?

@aamirshaikh7 is already keen to work on #877 🚀

**Comment by smlabt** [comment]
From my side it is ready to merge :)

**Pull request merged**